### PR TITLE
Options state refactor

### PR DESCRIPTION
**Pull request by BCrlrd**
This completely refactors the OptionsState.
The state is now split between the state itself (which just renders and updates) and its controller.
The coverage for both is 100%.

< organisation, testing

### CODE REVIEW

File: src/main/java/nl/tudelft/semgroup4/OptionsState.java [hunk 1]
```diff
@@ -26,95 +23,236 @@
...
 
-
-
     @Override
     public int getID() {
-        return States.OptionsState;
+        return controller.getId();
+    }
+
+    protected boolean getLoggerSetEnabled() {
```
Comment (Pathemeous) [inline src/main/java/nl/tudelft/semgroup4/OptionsState.java]: *Auto-generated comment*

    
    A getX() method which returns a boolean should be named isX()

File: src/main/java/nl/tudelft/semgroup4/OptionsStateController.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import org.newdawn.slick.GameContainer;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.Image;
+import org.newdawn.slick.Input;
+
+public class OptionsStateController {
+
+    private OptionsState state;
+    private ResourcesWrapper res;
+
```
Comment (Pathemeous) [inline src/main/java/nl/tudelft/semgroup4/OptionsStateController.java]: Both these fields could be final.

< naming

File: src/main/java/nl/tudelft/semgroup4/resources/ResourcesWrapper.java [hunk 3]
```diff
@@ -367,4 +367,12 @@ public Image createImage(int width, int height) throws SlickException {
     public TrueTypeFont createFont(Font font, boolean bool) {
         return new TrueTypeFont(font, bool);
     }
+
+    public boolean getMusicOn() {
```
Comment (Pathemeous) [inline src/main/java/nl/tudelft/semgroup4/resources/ResourcesWrapper.java]: *Auto-generated comment*

    
    A getX() method which returns a boolean should be named isX()

**Comment by Pathemeous** [comment]
100% coverage on a state, who would have thought we would manage that.

< testing, review.

**Comment by BCrlrd** [comment]
Everything has been renamed accordingly.

< naming

**Comment by BCrlrd** [comment]
Build failure seems to be caused by travis. Could anyone else have a look?

< review

**Comment by Pathemeous** [comment]
Triggered rebuild.

< review

**Comment by BCrlrd** [comment]
Well now it works.

< review

**Comment by Pathemeous** [comment]
Great, with the renames it all looks beautiful.

< review

**Pull request merged**